billboard.py

```diff
@@ -180,14 +168,14 @@ def fetchEntries(self):
         soup = BeautifulSoup(html, 'html.parser')
 
         prevWeek = soup.find('span', {'class': 'fa-chevron-left'})
-
-        if prevWeek:
-            prevLink = prevWeek.parent
-            if prevLink.get('href') is None:
-                self.date = '1958-08-04'
-            else:
-                self.previousDate = prevLink.get('href').split('/')[-1]
-                self.date = (datetime.datetime.strptime(self.previousDate, '%Y-%m-%d') + datetime.timedelta(days=7)).strftime('%Y-%m-%d') 
+        nextWeek = soup.find('span', {'class': 'fa-chevron-right'})
+        
+        if prevWeek and prevWeek.parent.get('href'):
+            self.previousDate = prevWeek.parent.get('href').split('/')[-1]
+            self.date = (datetime.datetime.strptime(self.previousDate, '%Y-%m-%d') + datetime.timedelta(days=7)).strftime('%Y-%m-%d') 
```

Comment:
The date should be parsed directly, not computed. It's wrong to assume that charts are always seven days apart. This is in fact not the case: the chart previous to 1962-01-06 was 1961-12-25. (This marked the switch from Monday-dated charts to Saturday-dated charts.)

Reply:
Tagging @mcookhome, since they introduced this logic.

Also, someone remind me to make the above example into a unit test.

Reply:
(Test added in 8a46ef083380f481f15f46f7b53f6af95f4a9025.)

Reply:
(Bug seemingly fixed by 57ce059.)

Reply:
Yep, that's my bad. Thanks for tagging me @guoguo12.

I only calculated it because I couldn't find anywhere on the page with the current date.

Reply:
glad it's fixed now

Reply:
@guoguo12 Good point; parsing the date directly just needed a different way to parse it given its format. Good that's fixed.